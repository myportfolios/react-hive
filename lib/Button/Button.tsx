import React from "react";

export function Button({ children }) {
  return <button>{children}</button>;
}
