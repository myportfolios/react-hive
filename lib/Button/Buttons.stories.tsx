import type { Meta, StoryObj } from "@storybook/react";
import { Button } from "./Button";

const meta: Meta<typeof Button> = {
  title: "Components/Button",
  component: Button,
  tags: ["autodocs"],
  parameters: {
    layout: "centered",
  },
  args: {
    /**methods to handle events on Button component*/
  },
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Default = {
  args: {
    children: "Primary Button",
  },
};
