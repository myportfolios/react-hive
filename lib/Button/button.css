@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .btn {
    @apply rounded bg-accent1 border border-secondary text-white;
  }
}

.btn-sm {
  @apply py-2 px-8;
}

.btn-md {
  @apply py-2 px-10;
}

.btn-lg {
  @apply py-2 px-12;
}
.btn-xl {
  @apply py-4 px-16;
}

.btn-text {
  @apply bg-transparent hover:bg-accent1 border-0 text-accent1 hover:text-white;
}

.btn-contained {
  @apply hover:bg-hover;
}
.btn-outlined {
  @apply bg-transparent border border-accent1 text-accent1 hover:border-2 font-bold;
}
