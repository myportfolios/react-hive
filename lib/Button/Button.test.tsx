import { render } from "@testing-library/react";
import { Button } from "./Button";

describe(`Component: ${Button.name}`, () => {
  it("should render", () => {
    const { container } = render(<Button>My Button</Button>);

    expect(container).toMatchSnapshot();
  });
  it("should match inner snapshot", () => {
    const { container } = render(<Button>My Button</Button>);

    expect(container).toMatchInlineSnapshot(`
      <div>
        <button
          class=" btn btn-md btn-text"
        >
          My Button
        </button>
      </div>
    `);
  });
});
