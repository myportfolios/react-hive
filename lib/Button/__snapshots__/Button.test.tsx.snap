// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: Button > should render 1`] = `
<div>
  <button
    class=" btn btn-md btn-text"
  >
    My Button
  </button>
</div>
`;
